gensim/test/test_corpora.py

```diff
@@ -225,6 +225,14 @@ def test_serialize_compressed(self):
         # MmCorpus needs file write with seek => doesn't support compressed output (only input)
         pass
 
+    def test_closed_file_object(self):
```

Comment:
need to add test for your context manager too (with file path (existent, non-existent), file-like object)

Reply:
Is it fine if I test whether the context manager throws exception or not for various cases ?

Reply:
I think yes